src/routes.js

```diff
@@ -44,4 +45,16 @@ router.post('/coffeegrams', async ctx => {
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
+router.post('/users/:username/bio', async ctx => {
+  var { bio } = ctx.request.body;
+  if (isBlank(bio)) {
+    ctx.request.flash('error', 'Your bio looks empty!');
+    return ctx.redirect('back');
+  }
+  var id = ctx.state.currentUser.id;
+  clean(bio);
+  await User.findByIdAndUpdate({ _id: id }, { $set: { bio }});
+  ctx.redirect(`/users/${ctx.state.currentUser.username}`);
```

Comment:
This implementation looks a bit sketchy. Your route allows a `:username`, but then your code doesn't use it and instead uses the currentUser. This is potentially dangerous behavior.

Instead, do this: the user being updated should be determined by the `:username`. Then, do an authorization check ("is the current user allowed to modify this?"), as opposed to authentication, which tells you "who is the current user?"

In your case, it's simple: if the `user` being edited is the `currentUser`, then they are allowed to modify the bio (since it's theirs). This is a common pattern: separating the "what are you modifying" (RESTful routes) from the "are you allowed to modify this" (authorization).